docs/components/TopLevelNav/TopLevelNav.tsx

```diff
@@ -134,16 +139,24 @@ const TopLevelNav = React.forwardRef((props: TopLevelNavProps, ref: React.Ref<HT
     }
   };
 
+  const ArrowIcon = () => {
+    if (rtl) {
+      return showSubmenu ? <AngleRight /> : <AngleLeft />;
+    } else {
+      return showSubmenu ? <AngleLeft /> : <AngleRight />;
+    }
+  };
```

Comment:
`ArrowIcon` is a function component, and in React, you cannot define another component within a component. You should define it outside the `TopLevelNav` component, otherwise, every time `TopLevelNav` renders, it will cause `ArrowIcon` to be redefined and recreated.

Reply:
Thank you for your advice. However, I would like to use the useMemo hook instead. Maybe the useMemo hook can help reduce the amount of code.